internal/plugin/data_plane_server.go

```diff
@@ -29,23 +30,29 @@ type (
 	}
 
 	DataPlaneServer struct {
-		address      string
-		logger       *slog.Logger
-		instances    []*instances.Instance
-		configEvents map[string][]*instances.ConfigurationStatus
-		messagePipe  bus.MessagePipeInterface
-		server       *http.Server
-		cancel       context.CancelFunc
+		address           string
+		logger            *slog.Logger
+		instances         []*instances.Instance
+		configEvents      map[string][]*instances.ConfigurationStatus
+		messagePipe       bus.MessagePipeInterface
+		server            *http.Server
+		cancel            context.CancelFunc
+		serverMutex       *sync.Mutex
+		configEventsMutex *sync.Mutex
+		instancesMutex    *sync.Mutex
 	}
 )
 
 func NewDataPlaneServer(agentConfig *config.Config, logger *slog.Logger) *DataPlaneServer {
 	address := net.JoinHostPort(agentConfig.DataPlaneAPI.Host, strconv.Itoa(agentConfig.DataPlaneAPI.Port))
 
 	return &DataPlaneServer{
-		address:      address,
-		logger:       logger,
-		configEvents: make(map[string][]*instances.ConfigurationStatus),
+		address:           address,
+		logger:            logger,
+		configEvents:      make(map[string][]*instances.ConfigurationStatus),
+		serverMutex:       &sync.Mutex{},
+		configEventsMutex: &sync.Mutex{},
+		instancesMutex:    &sync.Mutex{},
```

Comment:
Ditto here, do we expect the list of instances to change very often? I would think that the list of instances changes fairly rarely (writes are rare) but reads of the instances list to occur more regularly => `RWMutex` would be better.

Reply:
Sticking with Mutex unless we see a need for RWMutex